Orbitersdk/samples/ProjectApollo/src_sys/PanelSDK/Internals/esysparse.cpp

```diff
@@ -163,6 +163,80 @@ void E_system::Create_Diode(char *line)
 	AddSystem(new Diode(name, Source, NominalTemperature, SaturationCurrent));
 }
 
+void E_system::Create_ElectricLight(char* line)
+{
+	/* Config file creation string is as follows
+	
+	A single line begining with "<LIGHT>"
+
+	Followed by the fields:
+
+			Field			|			Unit			|			Type			|
+	---------------------------------------------------------------------------------
+		name						n/a							string
+		sourceName					n/a							string
+		flashing					n/a							string
+		OnTime						seconds						double
+		OffTime						seconds						double
+		<position>					meters						VECTOR3
+		<direction>					meters						VECTOR3
+		CutoffRange					meters						double
+		ConstantAttenuation			meters						double
+		LinearAttenuation			meters^-1					double
+		QuadraticAttenuation		meters^-2					double
+		Umbra						radians						double
+		Penumbra					radians						double
+		<%f %f %f %f>				n/a							COLOUR4
+		<%f %f %f %f>				n/a							COLOUR4
+		<%f %f %f %f>				n/a							COLOUR4
+		%lf							Watts						double
+		%lf							Volts						double
+	
+	
+	Example:
+
+	<LIGHT> SPOTLIGHT DC_DUMMY 0 0.0 0.0 <-1.439 1.390 0.920> <0.0045 -0.0046 0.9999> 5000.0 0.0 0.0 0.0015 0.174533 0.5 <1.0 0.945 0.878 0.0> <1.0 0.945 0.878 0.0> <0.0 0.0 0.0 0.0> 100.0 115.0
+
+
+	*/ 
+
+
+	char name[100];
+	char sourceName[100];
+	e_object* powerSource = nullptr;
+	bool flashing = false;
+	double onTime = 0;
+	double offTime = 0;
+	VECTOR3 pos = _V(0, 0, 0);
+	VECTOR3 dir = _V(1, 0, 0);
+	double range = 0;
+	double att0 = 0;
+	double att1 = 0;
+	double att2 = 0;
+	double umbra = 0;
+	double penumbra = 0;
+	COLOUR4 diffuse = { 0, 0, 0, 0};
+	COLOUR4 specular = { 0, 0, 0, 0 };
+	COLOUR4 ambient = { 0, 0, 0, 0 };
+	double powerDraw = 0;
+	double nomVoltage = 0;
+
+	sscanf(line + 7, "%s %s %d %lf %lf <%lf %lf %lf> <%lf %lf %lf> %lf %lf %lf %lf %lf %lf <%f %f %f %f> <%f %f %f %f> <%f %f %f %f> %lf %lf",
```

Comment:
sscanf doesn't have a specifier for booleans. So there either needs to be a temporary integer that then gets converted to the bool flashing or flashing needs to be an integer and then in the constructor for ElectricLight it converts to bool. As I said on the IRC, I got a runtime error in debug mode at the end of the function.

Reply:
this should also be fixed now.